res/res_pjsip_endpoint_identifier_ip.c

```diff
@@ -305,10 +305,37 @@ static struct ast_sip_endpoint *ip_identify(pjsip_rx_data *rdata)
 	return common_identify(ip_identify_match_check, &addr);
 }
 
+static struct ast_sip_endpoint *transport_identify(pjsip_rx_data *rdata)
```

Comment:
What happens given a scenario for a transport bound to any, and then another transport bound to an explicit address?

Reply:
You mean 0.0.0.0 as any? The code also matches the port number. I guess it would be user error if someone would try to do that.

One other thing related to this: The code should match the transport type as well. At least UDP vs TCP. In this case, TLS 
 has to be considered as TCP as well. Let me try to do that.

Reply:
As a user, how is it now configured and used? The documentation continues to confuse me.

Reply:
When a user configured asterisk with multiple ps_transports (let's say the box has multiple NICs or at least multiple IPs), the user can set identify_by=transport for the endpoint.

Then, specify a match=[one, or more of the bound IPs] in the ps_identify to match that endpoint.

Optionally, the user can also configure a transport=udp or a transport=tcp in the same ps_identify to match not just the bound IP (which is the Layer3 component of the TCP/IP stack), but also the transport protocol (which is Layer4 component of TCPI/IP stack).

This last, optional ps_identify attribute (transport=udp/tcp) is also available, when the endpoint is configured with identify_by=ip (the default, when the remote IP is matched).

Let me know if this explanation helps, then I'll try to incorporate this somehow in the documentation.

Reply:
Please consider the following scenario. The user has a **multi-tenant** asterisk instance with three customers (A, B and C). All three customer has a SIP trunk with the **same** ITSP. The ITSP addresses each customer on different IPs, for example

> customer A => 192.168.1.100
> customer B => 192.168.1.101
> customer C => 192.168.1.102

The ITSP has **just one address**: 192.168.1.200 and doesn't use any authentication, as this is on a dedicated private network. So _ip_ or _auth_username_ is not sufficient for endpoint identification _(identify_by)_. In this case the setup should look like this:

```
[transport-customer-a]
type=transport
bind=192.168.1.100:5060

[transport-customer-b]
type=transport
bind=192.168.1.101:5060

[transport-customer-c]
type=transport
bind=192.168.1.102:5060


[peer-customer-a]
type=endpoint
transport=transport-customer-a
identify_by=transport

[peer-customer-b]
type=endpoint
transport=transport-customer-b
identify_by=transport

[peer-customer-c]
type=endpoint
transport=transport-customer-c
identify_by=transport


[identity-customer-a]
type=identity
endpoint=peer-customer-a
match=192.168.1.100

[identity-customer-b]
type=identity
endpoint=peer-customer-b
match=192.168.1.101

[identity-customer-c]
type=identity
endpoint=peer-customer-c
match=192.168.1.102
```